### PR TITLE
docs: GridForm Documentation

### PR DESCRIPTION
**Pull request by songstephen**
### Overview

Added documentation, design information, and usage guidelines to the GridForm page in Storybook

### PR Checklist

- [X] Related to designs: https://www.figma.com/file/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=1689%3A3910
- [X] Related to JIRA ticket: [GM-260](https://codecademy.atlassian.net/browse/GM-260)
- [X] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx [hunk 1]
```diff
@@ -158,6 +153,39 @@ responsive layouts.
...
 </Canvas>
 
+## Designing with GridForm
+
+All [Form Input](https://www.figma.com/file/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=1189%3A0) components in the Figma library are consistent with their implementations in code. By setting the form inputs within the component's layout grid, we can design forms that are compatible with Gamut.
+
+The [GridForm page in Gamut](https://www.figma.com/file/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=1689%3A3910) also contains several starter templates for incorporating this organism in your designs.
+
+- **Starter**: Contains sample components to begin creating your own form
+- **Sections**: Contains the headers and dividers that are rendered in the optional [Sections](?id=organisms-gridform--submit-button-options&viewMode=docs#sections) modules
```
Comment (songstephen) [inline packages/styleguide/stories/Organisms/GridForm.stories.mdx:163]: Is this the proper way to link to an anchor on the same page in Storybook? Not sure if it'll scroll down to this, or if it'll refresh the page in some way.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@41.15.1-alpha.01e251.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f59212b66d4b37bb03cb30--gamut-preview.netlify.app>https://60f59212b66d4b37bb03cb30--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f59212b66d4b37bb03cb30>Deploy Logs</a></p>

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx (hunk 1 above)
Comment (dreamwasp) [reply packages/styleguide/stories/Organisms/GridForm.stories.mdx:163]: you can just do these as hash links like:

```suggestion
- **Sections**: Contains the headers and dividers that are rendered in the optional [Sections](#sections) modules
```

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx [hunk 2]
```diff
@@ -197,7 +225,7 @@ If an input is not meant to be usable, such as when a portion of a form is disab
 
 ## GridForm-atting
 
-We can use the `size` and `rowspan` props (borrowed from [LayoutGrid](https://react-hook-form.com)) to customize the layouts of our GridForms.
+We can use the `size` and `rowspan` props (borrowed from [LayoutGrid](/?path=/docs/layouts-layoutgrid--layout-grid)) to customize the layouts of our GridForms.
```
Comment (dreamwasp) [inline packages/styleguide/stories/Organisms/GridForm.stories.mdx:228]: 🙏

**Review by dreamwasp — COMMENTED** [review]
just a couple of tiny things but 🙌 🙌 🙌

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx (hunk 1 above)
Comment (codecaaron) [reply packages/styleguide/stories/Organisms/GridForm.stories.mdx:163]: You can also use the `<LinkTo>` component from the custom SB components we have:

```tsx
import { LinkTo } from '~styleguide/blocks';

<LinkTo id="foundations-color-mode--page">My Link Text</LinkTo>
```

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx [hunk 3]
```diff
@@ -158,6 +153,39 @@ responsive layouts.
...
+`GridForms` are laid out with `LayoutGrid`, so you can override its `columnGap` and `rowGap` with the normal string or responsive layouts to style and align the GridForm to your designs. You can also configure the `GridFormButtons` to modify the submit buttons and optional cancel buttons.
+
+While `Form` atoms compose the structure of these forms, we strongly recommend only creating forms with using `GridForm`. You should only use the form element atoms as individual elements if your interface doesn’t need to support the technical functions of a form.
+
+**You should build something outside of GridForm if you need...**
+
+- **Multi-step forms**: GridForm is best used for single page operations with a discrete submit action
+- **Custom behaviors for messaging or validation states**: form validation is already built into each input element, so any custom messaging is best done outside the component
+
 <PropsTable of={GridForm} />
```
[the remark this answers is not in the document]
Comment (songstephen) [reply packages/styleguide/stories/Organisms/GridForm.stories.mdx:189]: Makes sense! I'll keep the Stories and PropsTables together.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@41.15.1-alpha.09ef98.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f5e3897e1a3e214465e8c0--gamut-preview.netlify.app>https://60f5e3897e1a3e214465e8c0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f5e3897e1a3e214465e8c0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@42.1.1-alpha.c512d4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f872d31164f720af19118f--gamut-preview.netlify.app>https://60f872d31164f720af19118f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f872d31164f720af19118f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@42.1.2-alpha.01b017.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f99ddfc9bbf709456a42b5--gamut-preview.netlify.app>https://60f99ddfc9bbf709456a42b5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f99ddfc9bbf709456a42b5>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@42.1.3-alpha.e49dd0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60fb0d25598c6a0e8c507197--gamut-preview.netlify.app>https://60fb0d25598c6a0e8c507197--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60fb0d25598c6a0e8c507197>Deploy Logs</a></p>

**Pull request merged**